Tests/CryptomatorCloudAccessIntegrationTests/Box/BoxCredentialMock.swift

```diff
@@ -8,22 +8,48 @@
 
 import Foundation
 import Promises
-@testable import BoxSDK
+@testable import BoxSdkGen
 #if canImport(CryptomatorCloudAccessCore)
 @testable import CryptomatorCloudAccessCore
 #else
 @testable import CryptomatorCloudAccess
 #endif
 
+// Custom in-memory token storage
+class InMemoryTokenStore: TokenStorage {
+	private var tokenInfo: AccessToken?
+
+	func store(token: AccessToken) async throws {
+		tokenInfo = token
+	}
+
+	func get() async throws -> AccessToken? {
+		return tokenInfo
+	}
+
+	func clear() async throws {
+		tokenInfo = nil
+	}
+}
+
 class BoxCredentialMock: BoxCredential {
 	init() {
+		// Set up Box constants for testing purposes
 		BoxSetup.constants = BoxSetup(clientId: "", clientSecret: "", sharedContainerIdentifier: "")
-		super.init(tokenStore: MemoryTokenStore())
-		client = BoxSDK.getClient(token: IntegrationTestSecrets.boxDeveloperToken)
+
+		// Initialize the BoxCredential with InMemoryTokenStore
+		let tokenStore = InMemoryTokenStore()
+		super.init(tokenStore: tokenStore)
+
+		// Override the client with a test token using BoxDeveloperTokenAuth
+		let devTokenAuth = BoxDeveloperTokenAuth(token: IntegrationTestSecrets.boxDeveloperToken)
+		client = BoxClient(auth: devTokenAuth)
 	}
 
 	override func deauthenticate() -> Promise<Void> {
-		client = BoxSDK.getClient(token: "invalid")
+		// Set the client to an invalid token for deauthentication in tests
+		let invalidTokenAuth = BoxDeveloperTokenAuth(token: "invalid")
+		client = BoxClient(auth: invalidTokenAuth)
```

Comment:
I wonder if we should use a different `Authentication` implemention of BoxSdk, because we don't use the developer token in the main app, but an OAuth token.

If you compare [BoxDeveloperTokenAuth.swift](https://github.com/box/box-swift-sdk-gen/blob/62670ce06100cbabae3482b3a89c2365b6322d57/Sources/Box/DeveloperTokenAuth/BoxDeveloperTokenAuth.swift) with [BoxOAuth.swift](https://github.com/box/box-swift-sdk-gen/blob/62670ce06100cbabae3482b3a89c2365b6322d57/Sources/Box/Oauth/BoxOAuth.swift), you can see the different `BoxSDKErrors` that can be thrown.

I'm actually a little annoyed that you can't just catch the status code 401 for this.